notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java

```diff
@@ -100,13 +104,20 @@ public boolean removeTags(Collection<? extends String> tags) {
         return mTags.removeAll(tags);
     }
 
+    /**
+     * Adds several tamplates to this Installation.
+     * @param templates The templates to include with this Installation.
+     */
+    public void addTemplates(Map<String, InstallationTemplate> templates) { mTemplates.putAll(templates); }
+
     @Override
     public boolean equals(Object o) {

```

Comment:
Lets make sure to update the `hashCode()` method to include the templates as well. :)